dhctl/pkg/operations/bootstrap/steps.go

```diff
@@ -540,49 +540,79 @@ func setupRPPTunnel(sshClient *ssh.Client) (func(), error) {
 }
 
 func CheckDHCTLDependencies(nodeInteface node.Interface) error {
+
 	type checkResult struct {
 		name string
 		err  error
 	}
 
-	checkDependency := func(dep string) error {
+	checkDependency := func(dep string, resultsChan chan checkResult) error {
+
+		//breakPredicate := func(err error) bool {
+		//	var ee *exec.ExitError
+		//	if errors.As(err, &ee) {
+		//		if ee.ExitCode() == 255 {
+		//			return false
+		//		}
+		//	}
+		//	return true
+		//}
+
 		return retry.NewSilentLoop(fmt.Sprintf("Check dependency %s", dep), 30, 5*time.Second).Run(func() error {
 			output, err := nodeInteface.Command("command", "-v", dep).CombinedOutput()
+
 			if err != nil {
-				e := fmt.Errorf("bashible dependency error: %v - %s",
+				var ee *exec.ExitError
+				if errors.As(err, &ee) {
+					log.DebugF("exit code: %v", ee)
+				}
+				e := fmt.Errorf("bashible dependency %s error: %v - %s",
+					dep,
 					err,
 					string(output),
 				)
-
+				resultsChan <- checkResult{
+					name: dep,
+					err:  e,
+				}
 				log.DebugF("Dependency check error: %v\n", e)
-
 				return e
 			}
-
 			return nil
 		})
 	}
 
 	return log.Process("bootstrap", "Check DHCTL Dependencies", func() error {
-		dependencyArgs := []string{"sudo", "rm", "tar", "mount", "awk", "grep", "cut", "sed", "shopt",
+
+		dependencyCommands := []string{"sudo", "rm", "tar", "mount", "awk", "grep", "cut", "sed", "shopt",
```

Comment:
better way create slice of slices like 
```go
dependencyCommands := [][]string{
  { "sudo", "rm", "tar", "mount", "awk", },
  { "grep", "cut", "sed", "shopt", "mkdir", },
  { "cp", "join", "cat", "ps", "kill", },
}
```

and range through slices, than calculate indexes in `for` cycle

Reply:
done